README.md

```diff
@@ -16,3 +16,109 @@ The project and Python dependencies can be installed from source with:
+
+Global configuration are options that are user specific and stick with the user between projects, they are well suited for options like login credentials.
+The global configuration file defaults to `<platform-specific user configuration location>/edq-config.json`.
+The configuration location is chosen according to the [XDG standard](https://en.wikipedia.org/wiki/Freedesktop.org#Base_Directory_Specification) (implemented by [platformdirs](https://github.com/tox-dev/platformdirs)).
+Below are examples of user-specific configuration file paths for different operating systems:
+ - Linux -- `/home/<user>/.config/edq-config.json`
+ - Mac -- `/Users/<user>/Library/Application Support/edq-config.json`
+ - Windows -- `C:\Users\<user>\AppData\Local\edq-config.json`
+
+The default global configuration location can be changed by passing a path to `--config-global` through the command line.
+
+Below is an example command for specifying a global configuration path from the CLI:
+```sh
+python3 -m edq.cli.config.list --config-global  ~/.config/custom-config.json
+```
+
+#### Local Configuration
+Local configuration are options that are specific to a project or directory, they are useful for options relevant only to the work within that context.
+Local configuration files are searched in multiple locations, the first file found is used.
+The local config search order is:
+1. `edq-config.json` in the current directory.
+2. A legacy file in the current directory (only if a legacy file is preconfigured).
+3. `edq-config.json` in any ancestor directory on the path to root (including root itself).
+
+#### CLI-Specified Config Files
+CLI config files are options specified on the command line via a file, they are useful when you have a common set of options to apply but don't always apply for a user/project,
+for example when you want to login to an alternate user account.
+Any files passed via `--config-file` will be loaded in the order they appear on the command line.
+Options from later files override options from previous files.
+
+Below is an example of a CLI specified configuration paths:
+```sh
+python3 -m edq.cli.config.list --config-file ./edq-config.json --config-file ~/.secrets/edq-config.json
+```
+
+#### CLI Configuration
+CLI configurations are options specified directly on the command line, they are useful for quick configuration overrides without editing files.
+Configuration options are passed to the command line by the `--config` flag in this format `--config <key>=<value>`.
+The provided values overrides the values from configuration files.
+Configuration options are structured as key value pairs and keys cannot contain the "=" character.
+
+Below is an example of specifying a configuration option directly from the CLI:
+```sh
+python3 -m edq.cli.config.list --config user=alice --config pass=password123
+```
+
+#### CLI Config Options
+
+The table below lists all the common configuration CLI options available for CLI tools using this library.
```

Comment:
s/all the//

No need to be more specific here.
We don't want to make promises that could get us in trouble later.